polus-feature-subsetting-plugin/src/main.py

```diff
@@ -0,0 +1,258 @@
+import argparse, logging, subprocess, time, multiprocessing, sys
+import os
+import filepattern
+import pandas as pd
+import shutil
+from pathlib import Path
+import traceback
+
+def filter_planes(feature_dict, removeDirection, percentile):
+    """filter planes by the criteria specified by removeDirection
+    and percentile 
+
+    Args:
+        feature_dict (dictionary): planes and respective feature value
+        removeDirection (string): remove above or below percentile
+        percentile (int): cutoff percentile
+
+    Returns:
+        set: planes that fit the criteria
+    """
+    planes = list(feature_dict.keys())
+    feat_value = [feature_dict[i] for i in planes]
+    thresh = min(feat_value) + percentile * (max(feat_value) - min(feat_value))
+    
+    # filter planes
+    if removeDirection == 'Below':
+        keep_planes = [z for z in planes if feature_dict[z] >= thresh]
+    else:
+        keep_planes = [z for z in planes if feature_dict[z] <= thresh]
+    
+    return set(keep_planes)
+
+def make_uniform(planes_dict, uniques, delay):
+    """keep same number of planes across different sections
+    and capture additional planes based on specified delay
+
+    Args:
+        planes_dict (dict): planes to keep in different sections
+        uniques (list): unique values for the major grouping variable
+        delay (int): additional images to capture outside cutoff
+
+    Returns:
+        dictionary: dictionary containing planes to keep
+    """
+
+    # max no. of planes 
+    max_len = max([len(i) for i in planes_dict.values()])
+
+    # max planes that can be added on each side
+    min_ind = min([min(planes_dict[k]) for k in planes_dict])
+    max_ind = max([max(planes_dict[k]) for k in planes_dict])
+    max_add_left = uniques.index(min_ind)
+    max_add_right = len(uniques) - (uniques.index(max_ind)+1)
+    
+    # add planes in each section based on delay and max number of planes
+    for section_id, planes in planes_dict.items():
+        len_to_add = max_len - len(planes)
+        len_add_left = min(int(len_to_add)/2+delay, max_add_left)
+        len_add_right = min(len_to_add - len_add_left+delay, max_add_right)
+        left_ind = int(uniques.index(min(planes)) - len_add_left) 
+        right_ind = int(uniques.index(max(planes)) + len_add_right)+1
+        planes_dict[section_id] = uniques[left_ind:right_ind]
+    return planes_dict
+
+if __name__=="__main__":
```

Comment:
Something we are starting to transition to is putting any kind of functional processing of data into a `main()` function, and only parsing command line arguments under the `if __name__=='main'` block.

So, you should include all the logic to extract arguments under this section, but you should create a method called `main()` that receives those command line inputs. When you do this, make sure you use type definitions, and all input validation checks should be performed under the `main()` function.

For example, if you have a plugin with only an input and output directory, your `main()` function would look like this:

```python

def main(inpDir: pathlib.Path,
         outDir: pathlib.Path,
         ) -> None:
```